cli/provider_cache.go

```diff
@@ -234,7 +234,7 @@ func runTerraformCommand(ctx context.Context, opts *options.TerragruntOptions, a
 	errWriter := util.NewTrapWriter(opts.ErrWriter, HTTPStatusCacheProviderReg)
 
 	// add -no-color flag if the user specified it in the CLI
-	if util.ListContainsElement(opts.TerraformCliArgs, terraform.FlagNameNoColor) {
+	if !util.ListContainsElement(opts.TerraformCliArgs, terraform.FlagNameNoColor) {
```

Comment:
I think the comment above should be changed, otherwise it is confusing.

Reply:
Yes, updated

Reply:
Ah, I forgot, during normal operation, when the cache server returns a 423 code, we hide this error, but in all other cases we will display it as is. So it may happen that we will display this error without color, while in the rest of the logs the color will be.

Reply:
For example, if the provider's address contains an incorrect URL.